src/main/scala/gw/cda/api/outputwriter/OutputWriter.scala

```
@@ -26,46 +32,137 @@ trait OutputWriter {
-    * If there is a problem, throw an exception.
-    *
-    * In the case of local output, makes sure that the outputPath is a directory
-    * that exists and is not a file.
-    *
-    * In the case of S3 output, makes sure that the outputPath is in an existing
-    * S3 bucket and is also not an existing key to a S3 object.
-    *
-    * @throws java.io.IOException If the validation was not successful
-    *
-    */
+   * If there is a problem, throw an exception.
+   *
+   * In the case of local output, makes sure that the outputPath is a directory
+   * that exists and is not a file.
+   *
+   * In the case of S3 output, makes sure that the outputPath is in an existing
+   * S3 bucket and is also not an existing key to a S3 object.
+   *
+   * @throws java.io.IOException If the validation was not successful
+   *
+   */
   def validate(): Unit
 
-  /** Write a table and its schema to either local filesystem or to S3.
-    *
-    * @param tableDataFrameWrapperForMicroBatch has the data to be written
-    */
+  /** Write a table and its schema to either local filesystem or to S3
+   * and also to JDBC as indicated by clientConfig settings.
+   *
+   * @param tableDataFrameWrapperForMicroBatch has the data to be written
+   */
   def write(tableDataFrameWrapperForMicroBatch: DataFrameWrapperForMicroBatch): Unit = {
     val tableName = tableDataFrameWrapperForMicroBatch.tableName
-    log.info(s"Writing '$tableName' DataFrame as CSV to ${this.getPathToFolderWithCSV(tableDataFrameWrapperForMicroBatch)}")
-    this.writeCSV(tableDataFrameWrapperForMicroBatch)
-    this.writeSchema(tableDataFrameWrapperForMicroBatch)
-    log.info(s"Wrote '$tableName' DataFrame as CSV complete, with columns ${tableDataFrameWrapperForMicroBatch.dataFrame.columns.toList}")
+
+    // Process file write.
+    if (clientConfig.outputSettings.saveIntoFile) {
+      if (clientConfig.outputSettings.fileFormat.toLowerCase == "csv") {
```

I would write this out explicitly as a `match` statement. It's possible that the validation logic passes something that this code does not handle. Better to fail fast.
```scala
if (...) {
  clientConfig.outputSettings.fileFormat.toLowerCase match {
    case "csv" => ...
    case "parquet" => ...
    case other => throw new Exception(s"Unknown output file format $other")
  }
}
```

---

agreed. completed locally.